### ISSUE TITLE
windows.h breaks iwyu in certain cases

### ISSUE THREAD

**Issue by vsapsai** [include-what-you-use/include-what-you-use#183]
Originally reported on Google Code with ID 183
```
What steps will reproduce the problem?  Give the *exact* arguments passed
to include-what-you-use, and attach the input source file that gives the
problem (minimal test-cases are much appreciated!)
1. include windows.h before an associated header file for a c file
2. iwyu will no longer recognize the c file as requiring any function prototypes declared
in the associated header file

Arguments:
-Xiwyu --mapping_file=win32.imp -fmsc-version=1800 -ferror-limit=1 -Wno-deprecated-declarations
-Wno-format -Wno-incompatible-pointer-types -Wno-ignored-attributes -Wmissing-declarations
-Wimplicit-function-declaration -Werror

What is the expected output? What do you see instead?

Expected output:
(foo.h has correct #includes/fwd-decls)

(foo.c has correct #includes/fwd-decls)

Actual output:
foo.c should add these lines:

foo.c should remove these lines:
- #include "foo.h"  // lines 5-5

The full include-list for foo.c:
#include <windows.h>                    // for HANDLE

What version of the product are you using? On what operating system?
iwyu 0.3 compatible with llvm+clang 3.5, running on Windows 7 64-bit with VS2013

Please provide any additional information below.
To reproduce the problem, uncomment out the first line in foo.c (I've reduced all the
attached files to minimal test case). This is the only issue I've had with windows.h
on a fairly large codebase. (Everything other than prototypes seems to be picked up
okay)

```

Reported by `david.oman.86` on 2015-04-13 15:27:12

<hr>
 * *Attachment: [foo.c](https://storage.googleapis.com/google-code-attachments/include-what-you-use/issue-183/comment-0/foo.c)*
 * *Attachment: [foo.h](https://storage.googleapis.com/google-code-attachments/include-what-you-use/issue-183/comment-0/foo.h)*
 * *Attachment: [win32.imp](https://storage.googleapis.com/google-code-attachments/include-what-you-use/issue-183/comment-0/win32.imp)*

**Comment by kimgr** [comment]
I think this is the same as issue #5. Let's keep it around for now in case this is a smaller repro.

**Comment by kimgr** [comment]
Fixed by 23119ba.
